bin/i18n/sync-out.rb

```diff
@@ -30,6 +33,40 @@ def sync_out
   end
 end
 
+# Return true iff the specified file in the specified locale had changes
+# as of the most recent sync down.
+#
+# @param locale [String] the locale code to check. This can be either the
+#  four-letter code used internally (ie, "es-ES", "es-MX", "it-IT", etc), OR
+#  the two-letter code used by crowdin, for those languages for which we have
+#  only a single variation ("it", "de", etc).
+#
+# @param file [String] the path to the file to check. Note that this should be
+#  the relative path of the file as it exists within the locale directory; ie
+#  "/dashboard/base.yml", "/blockly-mooc/maze.json",
+#  "/course_content/2018/coursea-2018.json", etc.
+def file_changed?(locale, file)
+  @change_datas ||= CROWDIN_PROJECTS.keys.map do |crowdin_project|
+    project = Crowdin::Project.new(crowdin_project, nil)
```

Comment:
Is creating a Crowdin::Project expensive? Is it worth caching this object at all instead of creating it on every call to this file

Reply:
it's pretty cheap, but do note that `@change_datas` itself is cached